### PR TITLE
Utils for making eth2 networks

### PR DESCRIPTION
**Pull request by ralexstokes**
Part of the eth2 interop cleanup.

Grab bag of some utilities for doing various things to print genesis states or load genesis data.

A future PR will depend on this one.

#### Cute Animal Picture

put a cute animal picture link inside the parentheses

### CODE REVIEW

File: eth2/beacon/scripts/quickstart_state/generate_beacon_genesis.py [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+        pubkey=pubkey, withdrawal_credentials=withdrawal_credentials, amount=amount
+    )
+    deposit_data_signing_root = deposit_data.signing_root
+    signature = bls.sign(
+        deposit_data_signing_root,
+        privkey,
+        compute_domain(SignatureDomain.DOMAIN_DEPOSIT),
+    )
+
+    return deposit_data.copy(signature=signature)
```
Comment (hwwhww) [inline eth2/beacon/scripts/quickstart_state/generate_beacon_genesis.py]: Part of the logic could be refactored by reusing https://github.com/ethereum/trinity/blob/7b08d9d2bb140438e53f037585c4de31eff5c50f/eth2/beacon/tools/builder/validator.py#L715-L730

File: eth2/beacon/scripts/quickstart_state/generate_beacon_genesis.py [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    deposits = tuple()
+    for index, data in enumerate(deposit_datas):
+        length_mix_in = (index + 1).to_bytes(32, byteorder="little")
+        tree = calc_merkle_tree_from_leaves(deposit_data_leaves[: index + 1])
+
+        deposit = Deposit(
+            proof=(get_merkle_proof(tree, item_index=index) + (length_mix_in,)),
+            data=data,
+        )
+        deposits += (deposit,)
```
Comment (hwwhww) [inline eth2/beacon/scripts/quickstart_state/generate_beacon_genesis.py]: Most of it is similar to `create_mock_deposits_and_root` https://github.com/ethereum/trinity/blob/7065024d327e0c57d333203844e453beb0f089a3/eth2/beacon/tools/builder/initializer.py#L23-L77, maybe rework it to serve all cases?

File: eth2/beacon/scripts/quickstart_state/generate_beacon_genesis.py [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+    state = initialize_beacon_state_from_eth1(
+        eth1_block_hash=eth1_block_hash,
+        eth1_timestamp=eth1_timestamp,
+        deposits=deposits,
+        config=config,
+    )
+
+    # TODO make genesis_time configurable, ideally via some delay from now
+    the_state_we_want = state.copy(genesis_time=1567777777)
+    print(the_state_we_want.hash_tree_root)
```
Comment (hwwhww) [inline eth2/beacon/scripts/quickstart_state/generate_beacon_genesis.py]: `from eth_utils import encode_hex` and

```suggestion
    print(encode_hex(the_state_we_want.hash_tree_root))
```

File: eth2/beacon/scripts/quickstart_state/generate_beacon_genesis.py (hunk 1 above)
Comment (ralexstokes) [reply eth2/beacon/scripts/quickstart_state/generate_beacon_genesis.py]: yeah -- good call, i'll make a pass at merging this stuff before merging in this PR

the stuff in this PR is more or less "throw away" stuff we just needed for one-off tasks so didn't get a lot of love

**Pull request merged**